vamb/__main__.py

```diff
@@ -939,6 +980,306 @@ def run(
+
+    # Parse BAMs, save as npz
+    abundance = calc_rpkm(
+        abundance_options,
+        vamb_options.out_dir,
+        composition.metadata,
+        vamb_options.n_threads,
+        logfile,
+    )
+
+    time_generating_input = round(time.time() - begintime, 2)
+    log(
+        f"\nTNF and coabundances generated in {time_generating_input} seconds.",
+        logfile,
+        1,
+    )
+
+    tnfs, lengths = composition.matrix, composition.metadata.lengths
+    rpkms = abundance.matrix
+
+    if taxonomy_options.taxonomy_predictions_path:
+        df_gt = pd.read_csv(taxonomy_options.taxonomy_predictions_path)
+        graph_column = df_gt[f'predictions']
+        indices_mmseq = range(len(df_gt))
+    elif taxonomy_options.taxonomy_path and not taxonomy_options.no_predictor:
+        run_taxonomy_predictor(
+            vamb_options=vamb_options,
+            comp_options=comp_options,
+            abundance_options=abundance_options,
+            taxonomy_options=taxonomy_options,
+            predictor_training_options=predictor_training_options,
+            logfile=logfile,
+        )
+        PREDICTED_PATH = vamb_options.out_dir.joinpath("results_taxonomy_predictor.csv")
+        df_gt = pd.read_csv(PREDICTED_PATH)
+        graph_column = df_gt[f'predictions']
+        indices_mmseq = range(len(df_gt))
+    elif taxonomy_options.no_predictor:
+        df_mmseq = pd.read_csv(taxonomy_options.taxonomy_path, delimiter='\t', header=None)
+        df_mmseq = df_mmseq[~(df_mmseq[2] == 'no rank')]
+        graph_column = df_mmseq[8]
+        ind_map = {c: i for i, c in enumerate(composition.metadata.identifiers)}
+        indices_mmseq = [ind_map[c] for c in df_mmseq[0]]
+    else:
+        raise argparse.ArgumentTypeError('One of the taxonomy arguments is missing')
+
+    nodes, ind_nodes, table_parent = vamb.h_loss.make_graph(graph_column.unique())
+
+    classes_order = np.array(list(graph_column.str.split(';').str[-1]))
+    targets = [ind_nodes[i] for i in classes_order]
+
+    vae = vamb.h_loss.VAEVAEHLoss(
+        rpkms.shape[1], 
+        len(nodes), 
+        nodes, 
+        table_parent,
+        cuda=vamb_options.cuda,
+        logfile=logfile,
+    )
+
+    dataloader_vamb, mask_vamb = vamb.encode.make_dataloader(rpkms, tnfs, lengths)
+    dataloader_joint, mask_joint = vamb.h_loss.make_dataloader_concat_hloss(rpkms[indices_mmseq], tnfs[indices_mmseq], lengths[indices_mmseq], targets, len(nodes), table_parent)
+    dataloader_labels, mask_labels = vamb.h_loss.make_dataloader_labels_hloss(rpkms[indices_mmseq], tnfs[indices_mmseq], lengths[indices_mmseq], targets, len(nodes), table_parent)
+
+    shapes = (rpkms.shape[1], 103, 1, len(nodes))
+    dataloader = vamb.h_loss.make_dataloader_semisupervised_hloss(dataloader_joint, dataloader_vamb, dataloader_labels, len(nodes), table_parent, shapes)
+    MODEL_PATH = vamb_options.out_dir.joinpath("vaevae_model.pt")
```

Comment:
Please reserve capitalized identifier names for constants, following https://peps.python.org/pep-0008/#constants